src/libtrx/game/game_buf.c

```diff
@@ -1,52 +1,28 @@
 #include "game/game_buf.h"
 
 #include "enum_map.h"
-#include "game/shell.h"
-#include "log.h"
 #include "memory.h"
 
-static size_t m_Cap = 0;
-static size_t m_MemUsed = 0;
-static size_t m_MemFree = 0;
-static char *m_MemBase = nullptr;
-static char *m_MemPtr = nullptr;
+static MEMORY_ARENA_ALLOCATOR m_Allocator = {
+    .default_chunk_size = 1024 * 1024 * 5,
```

Comment:
This is 5 MB but your PR description says 10 MB.

Reply:
I updated it accordingly as 10 MB felt excessive.